### DISCUSSION TITLE
Ddt4all working / Pyren macro not working

### DISCUSSION THREAD

**Discussion by Marcovaldo78** [cedricp/ddt4all#1067] [Q&A]
Hello
I have read all the discussions and threads about Pyren and ddt4all…I changed my rlink display and now I have "radio not configured" message… Pyren recognize my ecus but when I apply macro gaves error "service not supported in active session" . I use USB elm with a modified can2 cable and I also changed parameters (min_tout)
I tried on PC with USB and wifi elm, changed USB to 230400, changed latency time
Also tried on Android phone (qpyren3)...same errors
My elm works on ddt4all and on Pyren, only macro commands gaves me those type of errors
Anyone can help me?
Twingo III - multimedia ecu on can2

Thank you very much

**Comment by KarelSvo** [comment]
Log files

**Comment by Marcovaldo78** [comment]
[ecu_log.txt](https://github.com/cedricp/ddt4all/files/14847672/ecu_log.txt)
[elm_log.txt](https://github.com/cedricp/ddt4all/files/14847673/elm_log.txt)
Wifi elm logs
Also with USB elm I have "service not supported" errors

I tried elm.py and init.txt modification as in Pyren issue 25 https://gitlab.com/py_ren/pyren/-/issues/25

**Comment by KarelSvo** [comment]
You send 1081 StopDiagSession  instead 10C0 StartDiagSession.

**Comment by Marcovaldo78** [comment]
I don't know...it's the macro from Pyren bot

**Comment by Marcovaldo78** [comment]
[cmd_can_10_1161_259209939R.txt](https://github.com/cedricp/ddt4all/files/14857610/cmd_can_10_1161_259209939R.txt)
Renamed from .CMD to txt

**Comment by shrlnm** [comment]
@Marcovaldo78 Open macro in text editor and replace 1081 on 10C0 or 1003

**Comment by Marcovaldo78** [comment]
[ecu_log.txt](https://github.com/cedricp/ddt4all/files/14863175/ecu_log.txt)
[elm_log.txt](https://github.com/cedricp/ddt4all/files/14863176/elm_log.txt)
[cmd_can_10_1161_259209939R.txt](https://github.com/cedricp/ddt4all/files/14863216/cmd_can_10_1161_259209939R.txt)

**Comment by Marcovaldo78** [comment]
MFD can is on can2, I am using modified cable (ddt4all recognize can2 ecus), maybe the problem could be the CMD the bot sent isn't can2??

**Comment by shrlnm** [comment]
CAN2 is about wiring. More over you receiving answers from ECU. 
Have you tried 10C0 for session opening?

**Comment by Marcovaldo78** [comment]
Yes both ... everything is working under can2 except macro...

**Comment by shrlnm** [comment]
@Marcovaldo78 Please share logs of ddt4all connection to the same ecu

**Comment by Marcovaldo78** [comment]
[ecu_ddt.txt](https://github.com/cedricp/ddt4all/files/14875821/ecu_ddt.txt)
[elm_ddt.txt](https://github.com/cedricp/ddt4all/files/14875822/elm_ddt.txt)

**Comment by KarelSvo** [comment]
>(https://github.com/cedricp/ddt4all/files/14863216/cmd_can_10_1161_259209939R.txt)

10C0 not 1003

`
  <EcuData name="11130" doc="MFD_X10_X98_0000_14_B" stdType="STD_B">
    <SessionId>genSession:GenSession?GenSession#11130</SessionId>
    <TrtId>diagServices:EcuDiag?EcuDiag#11130</TrtId>
    <ModelId>FG1011130.xml</ModelId>
    <OptimizerId>SG1011130.xml</OptimizerId>
    <ECUInformations>
      <Address targetAddress="57" toolAddress="F1" addressExtension="00" MType="0" TAType="0" />
      <Frames>
        <StartDiagSession request="10C0" response="50C0" mode="1" />
        <StopDiagSession request="1081" response="5081" mode="1" />
        <KeepAlive request="3E01" hasResponse="true" period="3000" negativeResponse="7F3E" mode="1" />
        <PlusAPCExtension request="" response="" mode="1" />
      </Frames>
      <Ids>
        <IdFrame request="2180" response="6180" length="0" mode="1" />
        <IdByte rank="6" mask="FF" value="14" />
      </Ids>
      <ErrorDelay value="1000" />
      <ReplyToRequestDelay value="0" />
      <CommRetry value="3" />
      <Programmable value="false" />
      <ProgrammableViaUSB value="true" mode="USB_FAT32" owner="CLIP" format="FAT32" />
      <CANConfig value="false" />
    </ECUInformations>
`

**Comment by Marcovaldo78** [comment]
Now with 10C0 ecu received commands in macro
But yet radio not configured...
[ecu_log.txt](https://github.com/cedricp/ddt4all/files/14876677/ecu_log.txt)
[elm_log.txt](https://github.com/cedricp/ddt4all/files/14876684/elm_log.txt)

**Comment by Marcovaldo78** [comment]
I noticed that macro command for vin writing was not successful  but on ddt4all I can see the same vin and crc present in other ecus

**Comment by shrlnm** [comment]
> But yet radio not configured..

Have you run the macro for 76?

**Comment by Marcovaldo78** [comment]
On 76 macro Is for Radio_dab unit right? Since I have changed only screen (MFD), do I need to send macro also for 76 ecu (radio dab)?  
Another question: on ACI I receive 3 data for 10 and 76 ecu, like 10:259204718R,259150423R,259150423R
When I send vin to pyren bot, which is the correct syntax? Like that : /vin VF1AH000561071161 10:259204718R,259150423R,259150423R ?
Thank you

**Comment by Marcovaldo78** [comment]
Great! After speaker test sound and pin request, radio is ALIVE! Now maps on SD isn't recognized...neither USB connection and voice control. Also, I can't fix hour and temperature.... Now I connect ddt4all and will see if I can turn on some function... 
Maybe for maps on SD I need Rlink explorer??
For now, thank you guys for all your support

**Comment by Marcovaldo78** [comment]
@KarelSvo @shrlnm  thank you for helping me! 👍
